src/batch_iterator.rs

```diff
@@ -129,25 +125,35 @@ where
     }
 }
 
+pub trait PointCloud: Sync {
+    type Id: ToString + Send;
+    type PointIter: Iterator<Item = Point>;
+    fn nodes_in_location(&self, query: &PointQuery) -> Vec<Self::Id>;
```

Comment:
Would it make sense to return an iterator here as well?

Reply:
Sure

Reply:
Then I have to specialize NodeIdsIterator too

Reply:
Actually, it becomes ugly because NodeIdsIterator borrows Octree, so you'd need to write `type PointIter = NodeIdsIterator<'a>`, but then you'd need to declare `'a` in the trait definition. Or use GATs, but they seem very unstable. I'd rather leave it like this for now.

Reply:
👍 Alright then, let's leave it as it is...